Update documentation (DG/PPP) (#190)
AY2223S1-CS2103-F14-3/tp: docs/DeveloperGuide.md

> * Data archiving of `Application` is done with adding a boolean attribute to the Application class as the record of its archive status.

Reviewer: ```suggestion
* Data archiving of `Application` is done by adding a boolean attribute to the Application class as a record of its archive status.
```

    * Pros: Reduced coupling between different classes.
    * Cons: Each time an action is performed on an application or interview (such as archiving, adding or editing), the list of upcoming interviews has to be informed and updated as well. 

### Statistic Feature

#### Implementation

> The statistic feature is a simple feature that allow user to obtain a summarized statistics of the whole application list.

Reviewer: ```suggestion
The statistic feature is a simple feature that allows users to obtain summarized statistics of the whole application list.
```

> The results of the statistic is shown on the UI using the `ResultDisplay` section. The sequence diagram below shows the working flow of statistic feature. It gets the list of applications in the model and count the respective information. The calculation result is then output through `CommandResult`.

Reviewer: ```suggestion
The summary statistics is shown on the UI using the `ResultDisplay` section. The sequence diagram below shows the workflow of the statistic feature. It gets the list of applications from the model and tabulates the respective information. The tabulation result is then output through `CommandResult`.
```

![Statistic Sequence Diagram](images/StatisticSequenceDiagram.png)

#### Constraints of Statistic Feature

> The statistic of the applications will only show when user enter `stats` command. Possible future improvement is to reorganise UI section to display real-time statistics on one section while the list view of applications and interviews are not affected.

Reviewer: ```suggestion
The statistic of the applications will only show when user enter `stats` command. Possible future improvement is to reorganise the UI section to display real-time statistics in one section and the list view of applications and interviews in another section.
```

#### Design Considerations

Aspect: How should the statistic feature be presented?

* Alternative 1 (current choice): Utilise `ResultDisplay` section in UI to show user the statistics.

>     * Pros: Does not need extra space in the UI to show the statistic and more simple and straight-forward implementation.

Reviewer: ```suggestion
    * Pros: Does not need extra space in the UI to show the statistic and hence simpler and more straightforward implementation.
```

    * Cons: Increase coupling between `ModelManager` and `StatsCommand` as it requires the list of applications in `ModelManager`.

* Alternative 2: Create a new section in UI to show user real-time statistics.
    * Pros: Does not increase coupling between classes and align with the implementation of Applications and Interviews list views.
    * Cons: Space usage for UI might be inefficient as user will not always want to review the statistic of the applications.

> - Alternative 1 is chosen as our team justified that the implementation is simple, and it will less likely to have bugs despite the accessing application list from `ModelManager`. Furthermore, interview and application list are more important to be shown in UI when comparing with the statistics.

Reviewer: ```suggestion
- Alternative 1 is chosen as our team justified that the implementation is simpler, and is less likely to contain bugs despite accessing application list from `ModelManager`. Furthermore, interview and application lists are more important to be shown on the GUI compared to the statistics.
```

1. _{ more test cases …​ }_

### Deleting an application

1. Deleting an application while all applications are being shown

   1. Prerequisites: List all applications using the `list` command. Multiple applications in the list.

   1. Test case: `delete 1`<br>

>       Expected: First application is deleted from the list. Details of the deleted application shown in the status message. Timestamp in the status bar is updated.

Reviewer: actually what is the `timestamp in the status bar` referring to? correct me if I am wrong but there's no such time stamp on the status bar right or am I looking at the wrong thing

Author: Oh yeah ps I didn't notice it, thanks for pointing out! 😅
